File: typespec/common/pagination.ts
export type PageSize = number;
export type PaginationKey = string;

export function isPageSize(value: PageSize): boolean {
  return Number.isInteger(value) && value >= 1 && value <= 100;
}

export function isPaginationKey(value: PaginationKey): boolean {
  const length = [...value].length;
  return length >= 1 && length <= 4096;
}

File: typespec/common/pagination.go
package common

import "unicode/utf8"

type PageSize int32
type PaginationKey string

func IsPageSize(value PageSize) bool {
	return value >= 1 && value <= 100
}

func IsPaginationKey(value PaginationKey) bool {
	length := utf8.RuneCountInString(string(value))
	return length >= 1 && length <= 4096
}
